providers/account.rb

```diff
@@ -234,3 +239,23 @@ def group_resource(exec_action)
       new_resource.updated_by_last_action(true) if r.updated_by_last_action?
   end
 end
+
+def keypair_resource(exec_action)
+  new_resource.ssh_keypair.each do |name, key|
+    # avoid variable scoping issues in resource block
+    key_name, key_content = name, key
+
+    home = Etc.getpwnam(new_resource.username).dir
```

Comment:
The problem with `@my_home` is, that it's just an educated guess, assuming that the home directory resides in `"#{node['user']['home_root']}/#{new_resource.username}"`.

I've just stumbled over the follwing code snipped in `account.rb` (which determines the gid), which seems to do exactly what we want:

https://github.com/chr4-cookbooks/user/blob/e4959685d6e5db96939c0fb0e6f51bc21901437d/providers/account.rb#L104-L110
```ruby
def user_gid
  # this check is needed as the new user won't exit yet
  # in why_run mode, causing an uncaught ArgumentError exception
  Etc.getpwnam(new_resource.username).gid
rescue ArgumentError
  nil
end
```

I'd suggest adding a `user_home` method in a similar fashion, which then should be used everywhere instead of `@my_home`. I'd use something like this:
```ruby
# Returns the user's home directory, either from the `home` attribute (if specified), or using Etc.getpwnam()
def user_home
  # This check is needed as the new user won't exit yet in why_run mode, 
  # causing an uncaught ArgumentError exception
  new_resource.home || Etc.getpwnam(new_resource.username).dir
rescue ArgumentError
  nil
end
```